model/DataObject.php

```diff
@@ -1926,13 +1926,21 @@ public function many_many_extraFields($component = null) {
 				$manyMany = $SNG_class->stat('belongs_many_many');
 				$candidate = (isset($manyMany[$component])) ? $manyMany[$component] : null;
 				if($candidate) {
+					$relationName = null;
+					// Extract class and relation name from dot-notation
+					if(strpos($candidate, '.') !== false) {
+						list($candidate, $relationName) = explode('.', $candidate);
+					}
+
 					$SNG_candidate = singleton($candidate);
 					$candidateManyMany = $SNG_candidate->stat('many_many');
 
 					// Find the relation given the class
-					if($candidateManyMany) foreach($candidateManyMany as $relation => $relatedClass) {
-						if($relatedClass == $class) {
-							$relationName = $relation;
+					if( ! $relationName && $candidateManyMany) {
+						foreach($candidateManyMany as $relation => $relatedClass) {
+							if($relatedClass == $class) {
```

Comment:
On line 2022 you're using: `$candidateClass == $class || is_subclass_of($class, $candidateClass)`, don't you need to do the same here?